Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,217 @@
+            {
+                if (!e.Message.Contains("The Application Default Credentials are not available"))
+                {
+                    Assert.Fail(String.Format("Expected InvalidOperationException containing message 'The Application Default Credentials are not available'. Obtained {0} instead.", e.ToString()));
+                }
+            }
+        }
+
+        // Environment variable points to a non existant credential file
+        [Test]
+        public void TestGetApplicationDefaultCredentials_MissingCredentialFile()
+        {
+            try
+            {
+                TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+                credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, WellKnownCredentialFilePath);
+
+                ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+                Assert.Fail(string.Format("No credential file specified. Test expected to result in exception. Obtained {0} instead.", credential.GetType()));
+            }
+            catch (InvalidOperationException e)
+            {
+                if (!e.Message.Contains("Please check the value of the Environment Variable GOOGLE_APPLICATION_CREDENTIALS"))
+                {
+                    Assert.Fail(String.Format("Expected InvalidOperationException containing message 'Please check the value of the Environment Variable GOOGLE_APPLICATION_CREDENTIALS'. Obtained {0} instead.", e.ToString()));
+                }
+            }
+        }
+
+        // Credential file has invalid content format
+        [Test]
+        public void TestGetApplicationDefaultCredentials_InvalidCredentialFile()
+        {
+            try
+            {
+                string credentialFilepath = "TempFilePath.json";
+                TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+                credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+                credentialProvider.SetFileContents(credentialFilepath, "Invalid Credentials File Contents");
+
+                ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+                Assert.Fail(string.Format("Invalid credential file specified. Test expected to result in exception. Obtained {0} instead.", credential.GetType()));
+            }
+            catch (InvalidOperationException e)
+            {
+                if (!e.Message.Contains("Error reading credential file from location"))
+                {
+                    Assert.Fail(String.Format("Expected InvalidOperationException containing message 'Error reading credential file from location'. Obtained {0} instead.", e.ToString()));
+                }
+            }
+        }
+
+
+        #endregion
+    }
+
+    class TestDefaultCredentialProvider : DefaultCredentialProvider
```

Comment:
Please add a comment and add it in the beginning of the file (or in a separated file)